test/support/generators.ex

```diff
@@ -0,0 +1,293 @@
+          data(depth + 1),
+          max_length: 5
+        )
+    end)
+    |> StreamData.one_of()
+  end
+
+  defp schematic_from_data(data) when is_integer(data),
+    do:
+      StreamData.member_of([
+        int(),
+        int(data),
+        oneof([int(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([int(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_binary(data),
+    do:
+      StreamData.member_of([
+        str(),
+        str(data),
+        oneof([str(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([str(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_boolean(data),
+    do:
+      StreamData.member_of([
+        bool(),
+        bool(data),
+        oneof([bool(), simple_schematic() |> Enum.fetch!(1)]),
+        oneof([bool(data), simple_schematic() |> Enum.fetch!(1)])
+      ])
+
+  defp schematic_from_data(data) when is_nil(data),
+    do: StreamData.member_of([null(), nullable(simple_schematic() |> Enum.fetch!(1))])
+
+  defp schematic_from_data(data) when is_tuple(data) do
+    Tuple.to_list(data)
+    |> Enum.map(fn datum -> schematic_from_data(datum) |> Enum.fetch!(1) end)
+    |> tuple()
+    |> StreamData.constant()
+  end
+
+  defp schematic_from_data(data) when is_list(data) do
+    one_of_schematic =
+      Enum.reduce(data, [], fn datum, acc ->
+        [schematic_from_data(datum) |> Enum.fetch!(1) | acc]
+      end)
+      |> oneof()
+      |> list()
+
+    StreamData.member_of([
+      list(),
+      one_of_schematic
+    ])
+  end
+
+  # TODO: generate schematics that allow for optional keys, and maybe add extra optional keys that aren't present in `data
+  defp schematic_from_data(data) when is_map(data) do
+    schematic_options =
+      Enum.reduce(data, {[], []}, fn {datum_key, datum_value}, {key_types, val_types} ->
+        {
+          [schematic_from_data(datum_key) |> Enum.fetch!(1) | key_types],
+          [schematic_from_data(datum_value) |> Enum.fetch!(1) | val_types]
+        }
+      end)
+      |> then(fn {key_types, val_types} ->
+        [key_schematic: oneof(key_types), value_schematic: oneof(val_types)]
+      end)
+      |> map()
+
+    blueprint =
+      Enum.reduce(data, %{}, fn {datum_key, datum_value}, acc ->
+        Map.put(acc, datum_key, schematic_from_data(datum_value) |> Enum.fetch!(1))
+      end)
+      |> map()
+
+    StreamData.member_of([schematic_options, blueprint])
+  end
+
+  @doc """
+  Generates structure-less data that has an analogous `Schematic` function. This excludes maps, lists, and tuples.
+  """
+  def scalar() do
+    StreamData.one_of([
+      StreamData.binary(),
+      StreamData.integer(),
+      StreamData.boolean(),
+      StreamData.constant(nil)
+    ])
+  end
+
+  @doc """
+  Generator for random, simple `Schematic`s. All `Schematic`s that this is capable of generating are
```

Comment:
an instance of a schematic should be referred to as a "schematic" and not a "Schematic"